marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/survivorship/merging/standard.xqy

```diff
@@ -132,235 +132,48 @@ declare function merging:standard-condense-properties(
 )
 {
   if (fn:count((
-        ($all-properties ! map:get(., "values") ! fn:root(.)) union ()
-      )) eq 1) then
+    ($all-properties ! map:get(., "values") ! fn:root(.)) union ()
+  )) eq 1) then
     $all-properties
   else
-    merging:merge-complementing-properties(
-      $all-properties,
-      ()
-    )
+    merging:merge-complementing-properties($all-properties)
 };
 
 declare function merging:merge-complementing-properties(
```

Comment:
The primary changes needed are in this function. Now it uses a hash to compare properties, instead of the previous recursive descent caparison that was so expensive